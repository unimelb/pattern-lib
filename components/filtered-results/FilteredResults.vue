<template>
  <div class="filtered-results">
    <div class="filtered-results__inner">
      <FilteredResultsTitle
        :items="items"
        :filters="filters"
        :callback="callback" />
      <div class="filtered-results__content">
        <div
          v-if="items.length"
          class="filtered-results__items">
          <slot />
        </div>
        <div
          v-else
          class="filtered-results__no-items">
          <h3>No results</h3>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import FilteredResultsTitle from './FilteredResultsTitle.vue';

export default {
  components: {
    FilteredResultsTitle,
  },
  props: {
    items: {
      type: Array,
      default: () => [],
      required: true,
    },
    filters: {
      type: Array,
      default: () => [],
      required: true,
    },
    callback: {
      type: Function,
      required: true,
      default: () => false,
    },
  },
};
</script>
