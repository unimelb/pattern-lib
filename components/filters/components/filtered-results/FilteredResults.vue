<template>
  <div class="filtered-results">
    <div class="filtered-results__inner">
      <FilteredResultsTitle
        :items="items"
        :filters="filters"
        :secondary-message="secondaryMessage"
        :callback="callback" />
      <div class="filtered-results__content">
        <div
          v-if="items > 0"
          class="filtered-results__items">
          <slot />
        </div>
        <div
          v-else
          class="filtered-results__no-items">
          <h3>No results</h3>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import FilteredResultsTitle from './FilteredResultsTitle.vue';

export default {
  components: {
    FilteredResultsTitle,
  },
  props: {
    items: {
      type: Number,
      default: 0,
      required: true,
    },
    filters: {
      type: Number,
      default: 0,
      required: true,
    },
    secondaryMessage: {
      type: String,
      default: () => '',
      required: false,
    },
    callback: {
      type: Function,
      default: () => false,
      required: false,
    },
  },
};
</script>
