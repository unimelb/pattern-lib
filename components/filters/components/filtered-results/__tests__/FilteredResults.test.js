import { mount, shallow } from 'vue-test-utils';

import FilteredResults from '../FilteredResults.vue';

describe('FilteredResults', () => {
  it('Should match snapshot: has results, has only items and filters props', () => {
    const result = mount(FilteredResults,
      {
        propsData: {
          items: 10,
          filters: 2,
        },
      }).element;
    expect(result).toMatchSnapshot();
  });

  it('Should match snapshot: has results, has items, filters and secondary message props', () => {
    const result = mount(FilteredResults,
      {
        propsData: {
          items: 10,
          filters: 2,
          secondaryMessage: 'Click here',
        },
      }).element;
    expect(result).toMatchSnapshot();
  });

  it('Should match snapshot: no results', () => {
    const result = mount(FilteredResults,
      {
        propsData: {
          items: 0,
          filters: 0,
          callback: () => false,
        },
      }).element;
    expect(result).toMatchSnapshot();
  });

  it('Should have default props and correct types', () => {
    const result = shallow(FilteredResults);
    const {
      items,
      filters,
      secondaryMessage,
      callback,
    } = result.vm.$options.props;
    expect(items.type).toBe(Number);
    expect(filters.type).toBe(Number);
    expect(callback.type).toBe(Function);
    expect(secondaryMessage.type).toBe(String);
  });
});
