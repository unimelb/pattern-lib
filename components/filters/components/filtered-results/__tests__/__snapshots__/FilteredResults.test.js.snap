// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilteredResults Should match snapshot: has results, has items, filters and secondary message props 1`] = `
<div
  class="filtered-results"
>
  <div
    class="filtered-results__inner"
  >
    <div>
      <h4
        class="filtered-results__title"
      >
        <span>
          
      10 results found with 
          <strong>
            2 filters
          </strong>
           applied.
    
        </span>
         
        <span
          class="filtered-results__title-notice text-italic text-bold"
        >
          Click here
        </span>
      </h4>
    </div>
     
    <div
      class="filtered-results__content"
    >
      <div
        class="filtered-results__items"
      />
    </div>
  </div>
</div>
`;

exports[`FilteredResults Should match snapshot: has results, has only items and filters props 1`] = `
<div
  class="filtered-results"
>
  <div
    class="filtered-results__inner"
  >
    <div>
      <h4
        class="filtered-results__title"
      >
        <span>
          
      10 results found with 
          <strong>
            2 filters
          </strong>
           applied.
    
        </span>
         
        <!---->
      </h4>
    </div>
     
    <div
      class="filtered-results__content"
    >
      <div
        class="filtered-results__items"
      />
    </div>
  </div>
</div>
`;

exports[`FilteredResults Should match snapshot: no results 1`] = `
<div
  class="filtered-results"
>
  <div
    class="filtered-results__inner"
  >
    <div>
      <h4
        class="filtered-results__title"
      >
        <span>
          
      0 results found with 
          <strong>
            0 filters
          </strong>
           applied.
    
        </span>
         
        <!---->
      </h4>
    </div>
     
    <div
      class="filtered-results__content"
    >
      <div
        class="filtered-results__no-items"
      >
        <h3>
          No results
        </h3>
      </div>
    </div>
  </div>
</div>
`;
